# Project 1: Deaths by tuberculosis

by Michel Wermelinger and Ivana Gramatikova 15 Decembre 2017,

This is the project notebook for Week 1 of The Open University's [_Learn to code for Data Analysis_](http://futurelearn.com/courses/learn-to-code) course.

In 2000, the United Nations set eight Millenium Development Goals (MDGs) to reduce poverty and diseases, improve gender equality and environmental sustainability, etc. Each goal is quantified and time-bound, to be achieved by the end of 2015. Goal 6 is to have halted and started reversing the spread of HIV, malaria and tuberculosis (TB).
TB doesn't make headlines like Ebola, SARS (severe acute respiratory syndrome) and other epidemics, but is far deadlier. For more information, see the World Health Organisation (WHO) page <http://www.who.int/gho/tb/en/>.

Given the population and number of deaths due to TB in some countries during one year, the following questions will be answered: 

- What is the total, maximum, minimum and average number of deaths in that year?
- Which countries have the most and the least deaths?
- What is the death rate (deaths per 100,000 inhabitants) for each country?
- Which countries have the lowest and highest death rate?

The death rate allows for a better comparison of countries with widely different population sizes.

## The data

The data consists of total population and total number of deaths due to TB (excluding HIV) in 2013 in each of the Balkan countries (Albania, Bosnia and Herzegovina, Serbia, Greece, Bulgaria, Croatia, Slovenia, Macedonia). 

The data was taken in July 2015 from <http://apps.who.int/gho/data/node.main.POP107?lang=en> (population) and <http://apps.who.int/gho/data/node.main.593?lang=en> (deaths). The uncertainty bounds of the number of deaths were ignored.

The data was collected into an Excel file which should be in the same folder as this notebook.

In [31]:
import warnings
warnings.simplefilter('ignore', FutureWarning)

from pandas import *
data = read_excel('WHO POP TB balkans.xls')
data

Country  Population (1000s)  TB deaths
0                                    Albania                3173         20
1                     Bosnia and Herzegovina                3829        190
2                                   Bulgaria                7223        150
3                                    Croatia                4290         52
4                                     Greece               11128         77
5                                     Serbia                9511        150
6                                   Slovenia                2072         21
7  The former Yugoslav republic of Macedonia                2107         33

## The range of the problem

The column of interest is the last one.

In [32]:
tbColumn = data['TB deaths']

The total number of deaths in 2013 is:

In [33]:
tbColumn.sum()

693

The largest and smallest number of deaths in a single country are:

In [34]:
tbColumn.max()

190

In [35]:
tbColumn.min()

20

From less than 20 to almost a quarter of a million deaths is a huge range. The average number of deaths, over all countries in the data, can give a better idea of the seriousness of the problem in each country.
The average can be computed as the mean or the median. Given the wide range of deaths, the median is probably a more sensible average measure.

In [36]:
tbColumn.mean()

86.625

In [37]:
tbColumn.median()

64.5

The median is far lower than the mean. This indicates that some of the countries had a very high number of TB deaths in 2013, pushing the value of the mean up.

## The most affected

To see the most affected countries, the table is sorted in ascending order by the last column, which puts those countries in the last rows.

In [38]:
data.sort('TB deaths')

Country  Population (1000s)  TB deaths
0                                    Albania                3173         20
6                                   Slovenia                2072         21
7  The former Yugoslav republic of Macedonia                2107         33
3                                    Croatia                4290         52
4                                     Greece               11128         77
2                                   Bulgaria                7223        150
5                                     Serbia                9511        150
1                     Bosnia and Herzegovina                3829        190

The table raises the possibility that a large number of deaths may be partly due to a large population. To compare the countries on an equal footing, the death rate per 100,000 inhabitants is computed.

In [39]:
populationColumn = data['Population (1000s)']
data['TB deaths (per 100,000)'] = tbColumn * 100 / populationColumn
data

Country  Population (1000s)  TB deaths  \
0                                    Albania                3173         20   
1                     Bosnia and Herzegovina                3829        190   
2                                   Bulgaria                7223        150   
3                                    Croatia                4290         52   
4                                     Greece               11128         77   
5                                     Serbia                9511        150   
6                                   Slovenia                2072         21   
7  The former Yugoslav republic of Macedonia                2107         33   

   TB deaths (per 100,000)  
0                 0.630318  
1                 4.962131  
2                 2.076699  
3                 1.212121  
4                 0.691948  
5                 1.577121  
6                 1.013514  
7                 1.566208

In [40]:
data.sort('TB deaths (per 100,000)')

Country  Population (1000s)  TB deaths  \
0                                    Albania                3173         20   
4                                     Greece               11128         77   
6                                   Slovenia                2072         21   
3                                    Croatia                4290         52   
7  The former Yugoslav republic of Macedonia                2107         33   
5                                     Serbia                9511        150   
2                                   Bulgaria                7223        150   
1                     Bosnia and Herzegovina                3829        190   

   TB deaths (per 100,000)  
0                 0.630318  
4                 0.691948  
6                 1.013514  
3                 1.212121  
7                 1.566208  
5                 1.577121  
2                 2.076699  
1                 4.962131

## Conclusions

The Balkan countries had a total of about 693 deaths due to TB in 2013. The median shows that half of these coutries had fewer than 64.5 deaths. The higher mean 86.625 indicates that some countries had higher number. The least affected were Albenia and Slovenia, with 21 and 21 deaths respectively, and the most affected were Serbia and Bosnia and Herxegovina with 150 and 190 deaths in a single year. However, taking the population size into account, the least affected were Albania and Greece with less than 0.7 deaths per 100 thousand inhabitants, and the most affected were Bulgaria and Bosnia and Herzegovina with over 2 deaths per 100,000 inhabitants.

One should not forget that most values are estimates, and that the chosen countries are a small sample of all the world's countries. Nevertheless, they convey the message that TB is still a major cause of fatalities, and that there is a huge disparity between countries, with several ones being highly affected.